### PR TITLE
Remove unnecessary length check for encap message size

### PR DESCRIPTION
**Pull request by jyao1**
Transport may append more bytes for alignment and cause failure. Also, the smaller header is already checked before. So just remove it here.

### CODE REVIEW

File: unit_test/test_spdm_requester/encap_request.c [hunk 1]
```diff
@@ -1079,14 +1079,10 @@ int libspdm_requester_encap_request_test_main(void)
         cmocka_unit_test(libspdm_test_requester_encap_request_case1),
         /* Success Case ,func :libspdm_get_encap_response_digest*/
         cmocka_unit_test(libspdm_test_requester_encap_request_case2),
-        /* Error response: When spdm_encapsulated_response_ack_response versions are inconsistent*/
```
Comment (steven-bellock) [inline unit_test/test_spdm_requester/encap_request.c:1082]: Why are these tests removed?

Comment (jyao1) [reply unit_test/test_spdm_requester/encap_request.c:1082]: The comment is completely wrong.

The test case just input large message size, and expect invalid msg size.

The SPM version is same.

Comment (steven-bellock) [reply unit_test/test_spdm_requester/encap_request.c:1082]: Please remove the code as well. For example : https://github.com/DMTF/libspdm/blob/6652fc3bbbe444e53637a22522a3fc9d45bd8c78/unit_test/test_spdm_requester/negotiate_algorithms.c#L1605-L1607

Comment (jyao1) [reply unit_test/test_spdm_requester/encap_request.c:1082]: done

**Pull request merged**